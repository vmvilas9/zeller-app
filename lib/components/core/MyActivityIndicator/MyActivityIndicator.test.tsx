import React from 'react';
import {render} from '@testing-library/react-native';
import {MyActivityIndicator} from './index';

describe('MyActivityIndicator', () => {
    it('renders correctly with default props', () => {
        const {toJSON} = render(<MyActivityIndicator/>);
        expect(toJSON()).toMatchSnapshot();
    });

    it('applies additional props passed to ActivityIndicator', () => {
        const {getByTestId} = render(
            <MyActivityIndicator testID="activity-indicator" color="#0000FF"/>
        );
        const indicator = getByTestId('activity-indicator');
        expect(indicator.props.color).toBe('#0000FF');
    });

    it('sets the size to large by default', () => {
        const {getByTestId} = render(<MyActivityIndicator testID="activity-indicator"/>);
        const indicator = getByTestId('activity-indicator');
        expect(indicator.props.size).toBe('large');
    });

    it('overrides the size when provided via props', () => {
        const {getByTestId} = render(
            <MyActivityIndicator testID="activity-indicator" size="small"/>
        );
        const indicator = getByTestId('activity-indicator');
        expect(indicator.props.size).toBe('small');
    });
});
