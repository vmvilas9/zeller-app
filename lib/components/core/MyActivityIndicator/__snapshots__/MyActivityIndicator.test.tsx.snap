// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyActivityIndicator renders correctly with default props 1`] = `
<ActivityIndicator
  size="large"
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`renders correctly 1`] = `
<ActivityIndicator
  size="large"
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
